tests/integration/pnl/test_historic_portfolio_volatility.py

```diff
@@ -0,0 +1,174 @@
+        rtn = np.random.multivariate_normal(
+            mean=self.mean, cov=self.cov, size=self.periods
+        )
+        df_returns = pd.DataFrame(
+            rtn, columns=[f"XX{ii}_AAXR" for ii in range(self.n_cids)], index=dates
+        )
+
+        return self._shape_df(df_returns)
+
+    def test_unit_weights(self):
+        # TODO signal types: [1] (1, 1, 1), [2] (0, 0, 0), [3] (1, 0, 0), [4] (0, 1, 0), [5] (0, 0, 1)
+        # TODO check logic of historic_portfolio_volatility follows above "contract signals"
+        # TODO monte-carlo logic checks on historic_portfolio_volatility
+        signal_types = [
+            (1, 1, 1),
+            (0, 0, 0),
+            (1, 0, 0),
+            (0, 1, 0),
+            (0, 0, 1),
+        ]
+
+        # Check portfolio volatility
+        dates = pd.bdate_range(
+            end=pd.Timestamp.today() + pd.offsets.BDay(n=0), periods=self.periods
+        )
+        start = dates.min()
+        end = dates.max()
+
+        def _get_test_df(dates, sig_type):
+            _dfx = pd.concat(
+                (
+                    self._gen_sig_df(dates=dates, sig_type=sig_type),
+                    self._gen_rtn_df(dates=dates),
+                ),
+                axis=0,
+            )
+            _dfx[["cid", "xcat"]] = _dfx.ticker.str.split("_", expand=True, n=1)
+            return _dfx
+
+        good_args = dict(
+            sname="STRAT",
+            rstring="XR",
+            est_freq="m",
+            lback_periods=21,
+            lback_meth="ma",
+            half_life=11,
+            start=None,
+            end=None,
+            blacklist=None,
+            nan_tolerance=0.25,
+            remove_zeros=True,
+            fids=[f"XX{ii}_AA" for ii in range(self.n_cids)],
+        )
+
+        rtn_df = self._gen_rtn_df(dates=dates)
+        sig_df = self._gen_sig_df(dates=dates, sig_type=signal_types[1])
+        df = pd.concat((rtn_df, sig_df), axis=0)
+
+        df_pvol = historic_portfolio_vol(df=df, **good_args)
+        self.assertIsInstance(df_pvol, QuantamentalDataFrame)
+        self.assertTrue(all(df_pvol["value"] == 0))
+
+        results = {}
+        for i in range(3):
+            rtn_df = self._gen_rtn_df(dates=dates)
+            sig_df = self._gen_sig_df(dates=dates, sig_type=signal_types[2 + i])
+            df_pvol = historic_portfolio_vol(
+                df=pd.concat((rtn_df, sig_df), axis=0), **good_args
+            )
+            results[i] = dict(df_pvol=df_pvol, rtn_df=rtn_df, sig_df=sig_df)
+            hvol = historic_vol(
+                df=rtn_df,
+                xcat="AAXR",
+                est_freq="m",
+                lback_periods=21,
+                lback_meth="ma",
+                cids=['XX' + str(list(signal_types[2 + i]).index(1))],
+                start=start.strftime("%Y-%m-%d"),
+                end=end.strftime("%Y-%m-%d"),
+            ).dropna().reset_index(drop=True)
+            hvol, df_pvol
+
+        results
```

Comment:
Why need to add the test of what value we expect - such as `(0, 0, 0)` being zero volatility for all periods.